### PR TITLE
NF-91 - Remove mistakenly uploaded docs

### PR DESCRIPTION
**Pull request by SteveSugden**

### CODE REVIEW

File: conf/app.routes [hunk 1]
```diff
@@ -27,6 +27,7 @@ GET         /upload-documents/error             uk.gov.hmrc.nationalimportdutyad
 
 GET         /uploaded-documents                 uk.gov.hmrc.nationalimportdutyadjustmentcentrefrontend.controllers.makeclaim.UploadFormSummaryController.onPageLoad()
 POST        /uploaded-documents                 uk.gov.hmrc.nationalimportdutyadjustmentcentrefrontend.controllers.makeclaim.UploadFormSummaryController.onSubmit()
+GET         /uploaded-documents/remove          uk.gov.hmrc.nationalimportdutyadjustmentcentrefrontend.controllers.makeclaim.UploadFormSummaryController.onRemove(documentReference: String)
```
Comment (crosslandwa) [inline conf/app.routes:30]: OBSERVATION: Should this be a HTTP DELETE? Its more accurate semantically, and will side-step any potential issues with returned cache headers

Comment (SteveSugden) [reply conf/app.routes:30]: I tried DELETE briefly and it didn't work (got 404 "Page not found").  I have a feeling that you can't make a DELETE request using an href in a tag element.

Ultimately (I think) this end-point will be used to render the "are you sure you want to remove?" page and the actual deleting will be done via a POST to the same endpoint.

File: app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/controllers/makeclaim/UploadFormSummaryController.scala [hunk 2]
```diff
@@ -52,7 +53,19 @@ class UploadFormSummaryController @Inject() (
...
+  def onRemove(documentReference: String): Action[AnyContent] = identify.async { implicit request =>
+    data.updateAnswers(removeDocument(documentReference)) map { _ =>
+      Redirect(controllers.makeclaim.routes.UploadFormSummaryController.onPageLoad())
+    }
   }
 
+  def removeDocument: String => UserAnswers => UserAnswers = (ref: String) =>
+    (userAnswers: UserAnswers) => {
+      val uploadedFiles  = userAnswers.uploads.getOrElse(Seq.empty)
+      val remainingFiles = uploadedFiles.filterNot(file => file.upscanReference == ref)
```
Comment (crosslandwa) [inline app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/controllers/makeclaim/UploadFormSummaryController.scala:67]: OBSERVATION: Not sure if this is any cleaner, but you could do away with the intermediate `uploadedFiles: Option[_]` via

```suggestion
      val remainingFiles = userAnswers.uploads.map(_.filterNot(_.upscanReference == ref))
```

and `remainingFiles` will already be an optional...

**Review by crosslandwa — APPROVED** [review]

File: app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/controllers/makeclaim/UploadFormSummaryController.scala (hunk 2 above)
Comment (SteveSugden) [reply app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/controllers/makeclaim/UploadFormSummaryController.scala]: Implemented your suggestion

**Review by crosslandwa — APPROVED** [review]

**Pull request merged**